test/tools/libtesteth/TestHelper.cpp

```diff
@@ -124,19 +125,126 @@ eth::Network stringToNetId(string const& _netname)
 json_spirit::mObject fillJsonWithTransaction(Transaction const& _txn)
 {
 	json_spirit::mObject txObject;
-	txObject["nonce"] = toCompactHex(_txn.nonce(), HexPrefix::Add, 1);
+	txObject["nonce"] = toCompactHex(_txn.nonce(), HexPrefix::Add, PrintZero::Print);
 	txObject["data"] = _txn.data().size() ? toHex(_txn.data(), 2, HexPrefix::Add) : "";
-	txObject["gasLimit"] = toCompactHex(_txn.gas(), HexPrefix::Add, 1);
-	txObject["gasPrice"] = toCompactHex(_txn.gasPrice(), HexPrefix::Add, 1);
-	txObject["r"] = toCompactHex(_txn.signature().r, HexPrefix::Add, 1);
-	txObject["s"] = toCompactHex(_txn.signature().s, HexPrefix::Add, 1);
-	txObject["v"] = toCompactHex(_txn.signature().v + 27, HexPrefix::Add, 1);
+	txObject["gasLimit"] = toCompactHex(_txn.gas(), HexPrefix::Add, PrintZero::Print);
+	txObject["gasPrice"] = toCompactHex(_txn.gasPrice(), HexPrefix::Add, PrintZero::Print);
+	txObject["r"] = toCompactHex(_txn.signature().r, HexPrefix::Add, PrintZero::Print);
+	txObject["s"] = toCompactHex(_txn.signature().s, HexPrefix::Add, PrintZero::Print);
+	txObject["v"] = toCompactHex(_txn.signature().v + 27, HexPrefix::Add, PrintZero::Print);
 	txObject["to"] = _txn.isCreation() ? "" : "0x" + toString(_txn.receiveAddress());
-	txObject["value"] = toCompactHex(_txn.value(), HexPrefix::Add, 1);
+	txObject["value"] = toCompactHex(_txn.value(), HexPrefix::Add, PrintZero::Print);
 	ImportTest::makeAllFieldsHex(txObject);
 	return txObject;
 }
 
+json_spirit::mObject fillJsonWithStateChange(State const& _stateOrig, eth::State const& _statePost, std::vector<dev::eth::detail::Change> const& _changeLog)
+{
+	json_spirit::mObject oState;
+	typedef dev::eth::detail::Change::Kind changeKind;
+	typedef std::vector<dev::eth::detail::Change> logVector;
+	std::vector<dev::eth::detail::Change> changeLog = _changeLog;
+
+	std::ostringstream log;
+	while(changeLog.size())
```

Comment:
If you need group changes by the addresses, you could sort `changeLog` vector by `address` field, then do all the processing in one iteration over the vector